### ISSUE TITLE
Added dependency on lack-middleware-session breaks quickload

### ISSUE THREAD

**Issue by m00natic** [40ants/weblocks#16]
Updating to latest ultralisp then trying to quickload :weblocks with
sbcl(-bin) 1.4.14 run into this:

> [package weblocks/server]
> debugger invoked on a SB-KERNEL:SIMPLE-PACKAGE-ERROR in thread
> #<THREAD "main thread" RUNNING {10005205B3}>:
>   The name "LACK-MIDDLEWARE-SESSION" does not designate any package.
> 
> Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.
> 
> restarts (invokable by number or by possibly-abbreviated name):
>   0: [RETRY                        ] Retry
>                                      compiling #<CL-SOURCE-FILE "weblocks/server" "lisp">.
>   1: [ACCEPT                       ] Continue, treating
>                                      compiling #<CL-SOURCE-FILE "weblocks/server" "lisp">
>                                      as having been successful.
>   2:                                 Retry ASDF operation.
>   3: [CLEAR-CONFIGURATION-AND-RETRY] Retry ASDF operation after resetting the
>                                      configuration.
>   4:                                 Retry ASDF operation.
>   5:                                 Retry ASDF operation after resetting the
>                                      configuration.
>   6: [ABORT                        ] Give up on "weblocks"
>   7:                                 Exit debugger, returning to top level.
> 
> (SB-INT:FIND-UNDELETED-PACKAGE-OR-LOSE "LACK-MIDDLEWARE-SESSION")

Reverting the latest commit e8b15e15c4f7b86157c35807bf56308b849dc510
fixes it for me.

By the way I've been using reblocks for the past 2 weeks for an
internal app and find it joyful.  And that not having dealt with web
for 10 years, let alone lisp web frameworks.  Thanks for trying to
relive it!

**Comment by vindarel** [comment]
\+1, was hit by this too. I asked on the gitter tchat, shame on me. Alexander fixed it with https://github.com/40ants/weblocks/commit/e8b15e15c4f7b86157c35807bf56308b849dc510

btw, nice to hear a feedback. Could we expect to see the sources someday ?

**Comment by m00natic** [comment]
Hm, actually what you cite as the fix has been the problem for me.  It
was OK before that.

Unfortunately the company I'm is not too FLOSS friendly and expect
this to stay internal as it builds on another internal library (CL
again!) which deals with sort of private customer data.  It's less
than 1000 lines of code currently (testament to the abstractions
Weblock provides), not too complex and would not be great loss.  In
any case, I've become a fan and will gladly use Weblocks for personal
stuff (when it comes) and will share.  Meanwhile I'll see if I can get
energy to go deeper in what's going with reblocks and possibly help.
No promises!

**Comment by vindarel** [comment]
oops for the commit :D 

Feel free to add your company to https://github.com/azzamsa/awesome-lisp-companies :) the more there are, the more attractive CL becomes (it's how things work!).

See you soon then :)

**Comment by svetlyak40wt** [comment]
Oh, I should just push to the main branch. Will make a pull request next time, and wait for approve at least from travis.

**Comment by svetlyak40wt** [comment]
By the way, it is interesting, but for now Travis says that tests are passing.

**Comment by svetlyak40wt** [comment]
@m00natic could you create a minimal reproducible example?

I tried to reproduce it in a clean environment, created with qlot.

I used these files:

### qlfile
```
dist ultralisp http://dist.ultralisp.org

ql :all :latest

ultralisp :all :latest
```

### qlfile.lock
```
("quicklisp" .
 (:class qlot.source.ql:source-ql-all
  :initargs (:distribution "http://beta.quicklisp.org/dist/quicklisp.txt" :%version :latest)
  :version "2018-10-18"))
("ultralisp" .
 (:class qlot.source.ql:source-ql-all
  :initargs (:distribution "http://dist.ultralisp.org" :%version :latest)
  :version "20181207-072339"))
```

Then launched the `qlot install`, started the emacs with repl, like that: `qlot exec ros emacs`, and in the REPL gave this command: `(ql:quickload :weblocks)`. And it loaded weblocks without any issues:

```
CL-USER> (ql:quickload :weblocks)
To load "weblocks":
  Load 2 ASDF systems:
    asdf uiop
  Install 76 Quicklisp releases:
    40ants-log4cl-json 40ants-weblocks alexandria anaphora
    asdf-system-connections babel bordeaux-threads cffi
    chipz chunga circular-streams cl+ssl cl-annot cl-base64
    cl-containers cl-cookie cl-fad cl-ppcre cl-reexport
    cl-routes cl-strings cl-syntax cl-utilities closer-mop
    dissect f-underscore fare-quasiquote fare-utils
    fast-http fast-io flexi-streams fukamachi-clack
    fukamachi-dexador fukamachi-lack fukamachi-quri
    global-vars http-body introspect-environment ironclad
    iterate jonathan lisp-namespace local-time log4cl
    metabang-bind metacopy metatilities metatilities-base
    moptilities named-readtables nibbles optima parenscript
    parse-declarations parse-number proc-parse puri salza2
    serapeum smart-buffer spinneret split-sequence
    static-vectors string-case trivia trivial-backtrace
    trivial-features trivial-file-size trivial-garbage
    trivial-gray-streams trivial-macroexpand-all
    trivial-mimes trivial-timeout trivial-types usocket
    xsubseq
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-timeout/2018-01-31/trivial-timeout-20180131-git.tgz">
; 8.36KB
==================================================
8,556 bytes in 0.00 seconds (8448.40KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-backtrace/2016-05-31/trivial-backtrace-20160531-git.tgz">
; 8.73KB
==================================================
8,941 bytes in 0.00 seconds (10507.15KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/parenscript/2016-03-18/Parenscript-2.6.tgz">
; 84.81KB
==================================================
86,845 bytes in 0.03 seconds (2695.02KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/spinneret/2018-08-31/spinneret-20180831-git.tgz">
; 23.48KB
==================================================
24,046 bytes in 0.11 seconds (207.82KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-macroexpand-all/2017-10-23/trivial-macroexpand-all-20171023-git.tgz">
; 1.92KB
==================================================
1,968 bytes in 0.00 seconds (1912.31KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-file-size/2018-01-31/trivial-file-size-20180131-git.tgz">
; 3.18KB
==================================================
3,254 bytes in 0.00 seconds (3395.02KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/lisp-namespace/2017-11-30/lisp-namespace-20171130-git.tgz">
; 9.53KB
==================================================
9,762 bytes in 0.00 seconds (9940.77KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivia/2018-10-18/trivia-20181018-git.tgz">
; 41.97KB
==================================================
42,974 bytes in 0.02 seconds (2242.18KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/string-case/2018-07-11/string-case-20180711-git.tgz">
; 8.87KB
==================================================
9,081 bytes in 0.00 seconds (9607.98KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/parse-number/2018-02-28/parse-number-v1.7.tgz">
; 5.58KB
==================================================
5,715 bytes in 0.00 seconds (4840.46KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/parse-declarations/2010-10-06/parse-declarations-20101006-darcs.tgz">
; 35.80KB
==================================================
36,664 bytes in 0.11 seconds (314.08KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/introspect-environment/2015-10-31/introspect-environment-20151031-git.tgz">
; 8.91KB
==================================================
9,127 bytes in 0.00 seconds (8807.40KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/global-vars/2014-11-06/global-vars-20141106-git.tgz">
; 3.50KB
==================================================
3,581 bytes in 0.00 seconds (4355.01KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/optima/2015-07-09/optima-20150709-git.tgz">
; 19.87KB
==================================================
20,345 bytes in 0.11 seconds (177.82KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/fare-utils/2017-01-24/fare-utils-20170124-git.tgz">
; 31.84KB
==================================================
32,604 bytes in 0.00 seconds (10767.62KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/fare-quasiquote/2017-11-30/fare-quasiquote-20171130-git.tgz">
; 15.74KB
==================================================
16,113 bytes in 0.00 seconds (13472.05KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/serapeum/2018-10-18/serapeum-20181018-git.tgz">
; 170.97KB
==================================================
175,075 bytes in 0.33 seconds (514.07KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/salza2/2013-07-20/salza2-2.0.9.tgz">
; 15.16KB
==================================================
15,525 bytes in 0.00 seconds (15630.03KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-routes/2017-01-24/cl-routes-20170124-git.tgz">
; 16.56KB
==================================================
16,959 bytes in 0.00 seconds (12700.55KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/puri/2018-02-28/puri-20180228-git.tgz">
; 29.30KB
==================================================
30,002 bytes in 0.11 seconds (265.73KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/metabang-bind/2017-11-30/metabang-bind-20171130-git.tgz">
; 22.12KB
==================================================
22,655 bytes in 0.00 seconds (19931.55KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/metatilities-base/2017-04-03/metatilities-base-20170403-git.tgz">
; 68.61KB
==================================================
70,258 bytes in 0.03 seconds (2239.93KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-containers/2017-04-03/cl-containers-20170403-git.tgz">
; 223.91KB
==================================================
229,279 bytes in 0.10 seconds (2290.19KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/asdf-system-connections/2017-01-24/asdf-system-connections-20170124-git.tgz">
; 4.82KB
==================================================
4,933 bytes in 0.00 seconds (1583.62KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/metatilities/2018-02-28/metatilities-20180228-git.tgz">
; 189.85KB
==================================================
194,405 bytes in 0.06 seconds (3271.67KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/moptilities/2017-04-03/moptilities-20170403-git.tgz">
; 15.25KB
==================================================
15,618 bytes in 0.11 seconds (136.70KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/metacopy/2017-04-03/metacopy-20170403-darcs.tgz">
; 8.97KB
==================================================
9,186 bytes in 0.01 seconds (1448.52KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/iterate/2018-02-28/iterate-20180228-git.tgz">
; 325.97KB
==================================================
333,795 bytes in 0.23 seconds (1415.82KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/dissect/2018-08-31/dissect-20180831-git.tgz">
; 28.89KB
==================================================
29,580 bytes in 0.00 seconds (14618.78KB/sec)
; Fetching #<URL "http://dist.ultralisp.org/ultralisp/archive/40ants-log4cl-json-20181124183503.tgz">
; 31.51KB
==================================================
32,263 bytes in 0.11 seconds (294.41KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/log4cl/2018-10-18/log4cl-20181018-git.tgz">
; 891.82KB
==================================================
913,227 bytes in 0.16 seconds (5540.58KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/http-body/2016-12-04/http-body-20161204-git.tgz">
; 14.62KB
==================================================
14,967 bytes in 0.01 seconds (2407.55KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/circular-streams/2016-12-04/circular-streams-20161204-git.tgz">
; 3.26KB
==================================================
3,335 bytes in 0.00 seconds (4431.07KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-types/2012-04-07/trivial-types-20120407-git.tgz">
; 3.15KB
==================================================
3,228 bytes in 0.00 seconds (3825.66KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/named-readtables/2018-01-31/named-readtables-20180131-git.tgz">
; 32.10KB
==================================================
32,869 bytes in 0.01 seconds (3675.98KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-syntax/2015-04-07/cl-syntax-20150407-git.tgz">
; 3.03KB
==================================================
3,102 bytes in 0.00 seconds (4352.44KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-annot/2015-06-08/cl-annot-20150608-git.tgz">
; 9.80KB
==================================================
10,039 bytes in 0.00 seconds (9179.50KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/jonathan/2018-04-30/jonathan-20180430-git.tgz">
; 156.27KB
==================================================
160,021 bytes in 0.57 seconds (276.10KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/f-underscore/2010-10-06/f-underscore-20101006-darcs.tgz">
; 0.96KB
==================================================
982 bytes in 0.00 seconds (1503.11KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/usocket/2018-08-31/usocket-0.7.1.tgz">
; 76.30KB
==================================================
78,129 bytes in 0.24 seconds (322.10KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-mimes/2018-08-31/trivial-mimes-20180831-git.tgz">
; 22.07KB
==================================================
22,597 bytes in 0.00 seconds (17853.87KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/static-vectors/2017-10-19/static-vectors-v1.8.3.tgz">
; 7.05KB
==================================================
7,218 bytes in 0.00 seconds (7780.16KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/fast-io/2017-10-23/fast-io-20171023-git.tgz">
; 7.88KB
==================================================
8,067 bytes in 0.00 seconds (8891.57KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/xsubseq/2017-08-30/xsubseq-20170830-git.tgz">
; 3.91KB
==================================================
4,006 bytes in 0.00 seconds (4313.24KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/smart-buffer/2016-06-28/smart-buffer-20160628-git.tgz">
; 3.10KB
==================================================
3,176 bytes in 0.00 seconds (2999.58KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/fast-http/2018-08-31/fast-http-20180831-git.tgz">
; 32.73KB
==================================================
33,518 bytes in 0.00 seconds (9376.23KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-reexport/2015-07-09/cl-reexport-20150709-git.tgz">
; 2.56KB
==================================================
2,621 bytes in 0.00 seconds (1974.98KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/split-sequence/2018-10-18/split-sequence-v1.5.0.tgz">
; 4.50KB
==================================================
4,611 bytes in 0.00 seconds (5140.33KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-utilities/2010-10-06/cl-utilities-1.2.4.tgz">
; 22.46KB
==================================================
22,998 bytes in 0.00 seconds (10297.56KB/sec)
; Fetching #<URL "http://dist.ultralisp.org/ultralisp/archive/fukamachi-quri-20181116095616.tgz">
; 262.78KB
==================================================
269,082 bytes in 0.20 seconds (1313.93KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/proc-parse/2016-03-18/proc-parse-20160318-git.tgz">
; 7.72KB
==================================================
7,905 bytes in 0.00 seconds (9722.58KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/local-time/2018-02-28/local-time-20180228-git.tgz">
; 291.06KB
==================================================
298,048 bytes in 0.23 seconds (1278.54KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-cookie/2015-08-04/cl-cookie-20150804-git.tgz">
; 4.75KB
==================================================
4,861 bytes in 0.00 seconds (5644.55KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-base64/2015-09-23/cl-base64-20150923-git.tgz">
; 8.39KB
==================================================
8,589 bytes in 0.00 seconds (10216.44KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-garbage/2018-10-18/trivial-garbage-20181018-git.tgz">
; 9.90KB
==================================================
10,136 bytes in 0.00 seconds (10100.45KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/flexi-streams/2018-07-11/flexi-streams-20180711-git.tgz">
; 129.46KB
==================================================
132,563 bytes in 0.05 seconds (2697.56KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cffi/2018-02-28/cffi_0.19.0.tgz">
; 250.25KB
==================================================
256,251 bytes in 0.23 seconds (1106.43KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl+ssl/2018-10-18/cl+ssl-20181018-git.tgz">
; 45.21KB
==================================================
46,294 bytes in 0.02 seconds (1860.38KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-gray-streams/2018-10-18/trivial-gray-streams-20181018-git.tgz">
; 7.84KB
==================================================
8,024 bytes in 0.00 seconds (8735.72KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/chunga/2018-01-31/chunga-20180131-git.tgz">
; 20.07KB
==================================================
20,552 bytes in 0.00 seconds (8507.97KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/chipz/2018-03-28/chipz-20180328-git.tgz">
; 35.87KB
==================================================
36,729 bytes in 0.02 seconds (1667.98KB/sec)
; Fetching #<URL "http://dist.ultralisp.org/ultralisp/archive/fukamachi-dexador-20181007062549.tgz">
; 739.88KB
==================================================
757,637 bytes in 1.24 seconds (595.65KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/closer-mop/2018-10-18/closer-mop-20181018-git.tgz">
; 22.16KB
==================================================
22,694 bytes in 0.11 seconds (195.49KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/nibbles/2018-08-31/nibbles-20180831-git.tgz">
; 17.23KB
==================================================
17,647 bytes in 0.11 seconds (156.08KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/ironclad/2018-10-18/ironclad-v0.43.tgz">
; 1367.92KB
==================================================
1,400,749 bytes in 0.38 seconds (3642.57KB/sec)
; Fetching #<URL "http://dist.ultralisp.org/ultralisp/archive/fukamachi-lack-20181109101309.tgz">
; 538.91KB
==================================================
551,845 bytes in 0.49 seconds (1103.16KB/sec)
; Fetching #<URL "http://dist.ultralisp.org/ultralisp/archive/fukamachi-clack-20180924145627.tgz">
; 1772.79KB
==================================================
1,815,334 bytes in 0.87 seconds (2042.81KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-strings/2018-01-31/cl-strings-20180131-git.tgz">
; 10.82KB
==================================================
11,078 bytes in 0.00 seconds (11448.00KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-ppcre/2018-08-31/cl-ppcre-20180831-git.tgz">
; 151.37KB
==================================================
155,003 bytes in 0.34 seconds (451.17KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-fad/2018-04-30/cl-fad-20180430-git.tgz">
; 24.03KB
==================================================
24,609 bytes in 0.00 seconds (14954.71KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/bordeaux-threads/2018-07-11/bordeaux-threads-v0.8.6.tgz">
; 21.43KB
==================================================
21,941 bytes in 0.00 seconds (20175.86KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-features/2016-12-04/trivial-features-20161204-git.tgz">
; 10.40KB
==================================================
10,654 bytes in 0.00 seconds (9562.77KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/babel/2017-12-27/babel-20171227-git.tgz">
; 247.98KB
==================================================
253,935 bytes in 0.34 seconds (721.42KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/anaphora/2018-02-28/anaphora-20180228-git.tgz">
; 6.01KB
==================================================
6,151 bytes in 0.00 seconds (4976.67KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/alexandria/2017-08-30/alexandria-20170830-git.tgz">
; 49.93KB
==================================================
51,131 bytes in 0.02 seconds (2031.85KB/sec)
; Fetching #<URL "http://dist.ultralisp.org/ultralisp/archive/40ants-weblocks-20181206174624.tgz">
; 6570.26KB
==================================================
6,727,945 bytes in 1.16 seconds (5675.70KB/sec)
; Loading "weblocks"
..................................................
[package f-underscore]............................
[package metabang.utilities]......................
[package closer-mop]..............................
[package closer-common-lisp]......................
[package closer-common-lisp-user].................
[package metabang.moptilities]....................
[package metabang.cl-containers]..................
[package metabang.bind]...........................
[package metabang.bind.developer].................
[package weblocks/app-mop]........................
[package cl-ppcre]................................
[package anaphora]................................
[package anaphora-basic]..........................
[package anaphora-symbol].........................
[package editor-hints.named-readtables]...........
[package parenscript].............................
[package ps-js-symbols]...........................
[package ps-dom1-symbols].........................
[package ps-dom2-symbols].........................
[package ps-window-wd-symbols]....................
[package ps-dom-nonstandard-symbols]..............
[package ps-dhtml-symbols]........................
[package ps-js]...................................
[package alexandria.0.dev]........................
[package global-vars].............................
[package trivia.level0]...........................
[package trivia.level1]...........................
[package trivia.fail].............................
[package trivia.skip].............................
[package trivia.next].............................
[package trivia.level1.impl]......................
[package lisp-namespace]..........................
[package trivia.level2]...........................
[package trivia.level2.impl]......................
[package fare-stateful]...........................
[package fare-quasiquote].........................
[package trivia.quasiquote.impl]..................
[package split-sequence]..........................
[package string-case].............................
[package org.mapcar.parse-number].................
[package trivial-garbage].........................
[package bordeaux-threads]........................
[package optima.core].............................
[package optima]..................................
[package optima.extra]............................
[package tcr.parse-declarations-1.0]..............
[package introspect-environment]..................
[package trivial-file-size].......................
[package trivial-macroexpand-all].................
[package serapeum]................................
[package serapeum-user]...........................
[package serapeum/op].............................
[package serapeum/vector=]........................
[package serapeum/mop]............................
[package serapeum/internal-definitions]...........
[package serapeum/dispatch-case]..................
[package serapeum/generalized-arrays].............
[package impl-specific-gray]......................
[package trivial-gray-streams]....................
[package spinneret]...............................
[package spinneret-user]..........................
[package spinneret.tag]...........................
[package weblocks/html]...........................
[package weblocks/js/base]........................
[package cl-fad]..................................
[package path]....................................
[package cl-fad-ccl]..............................
[package weblocks/utils/string]...................
[package weblocks/utils/list].....................
[package weblocks/variables]......................
[package log4cl.package-setup]....................
[package log4cl-impl].............................
[package log].....................................
[package weblocks/app]............................
[package salza2]..................................
[package nibbles].................................
[package ironclad]................................
[package babel-encodings].........................
[package babel]...................................
[package weblocks/utils/misc].....................
[package puri]....................................
[package iterate].................................
[package routes]..................................
[package weblocks/routes].........................
[package weblocks/app-actions]....................
[package metacopy]................................
[package cl-utilities]............................
[package quri.port]...............................
[package quri.uri]................................
[package quri.encode].............................
[package quri.error]..............................
[package quri.util]...............................
[package quri.decode].............................
[package quri.uri.ftp]............................
[package quri.uri.http]...........................
[package quri.uri.ldap]...........................
[package quri.uri.file]...........................
[package quri.etld]...............................
[package quri.domain].............................
[package quri.parser].............................
[package quri]....................................
[package proc-parse]..............................
[package xsubseq].................................
[package flexi-streams]...........................
[package smart-buffer]............................
[package fast-http.http]..........................
[package fast-http.error].........................
[package fast-http.byte-vector]...................
[package fast-http.util]..........................
[package fast-http.parser]........................
[package fast-http.multipart-parser]..............
[package fast-http]...............................
[package trivial-types]...........................
[package cl-syntax]...............................
[package cl-annot.util]...........................
[package cl-annot.core]...........................
[package cl-annot.expand].........................
[package cl-annot.syntax].........................
[package cl-annot.helper].........................
[package cl-annot]................................
[package cl-annot.std]............................
[package cl-annot.eval-when]......................
[package cl-annot.doc]............................
[package cl-annot.class]..........................
[package cl-annot.slot]...........................
[package cffi-sys]................................
[package cffi-callbacks]..........................
[package cffi]....................................
[package cffi-features]...........................
[package static-vectors]..........................
[package fast-io].................................
[package jonathan.error]..........................
[package jonathan.util]...........................
[package jonathan.encode].........................
[package jonathan.decode].........................
[package jonathan.helper].........................
[package jonathan]................................
[package http-body.util]..........................
[package http-body.multipart].....................
[package http-body.json]..........................
[package http-body.urlencoded]....................
[package http-body]...............................
[package circular-streams]........................
[package lack.request]............................
[package weblocks/utils/uri]......................
[package weblocks/session]........................
[package weblocks/request]........................
[package lack.util]...............................
[package weblocks/actions]........................
[package weblocks/hooks]..........................
[package weblocks/commands].......................
[package weblocks/response].......................
[package usocket].................................
[package chunga]..................................
[package local-time]..............................
[package cl-cookie]...............................
[package trivial-mimes]...........................
[package chipz]...................................
[package cl-base64]...............................
[package cl-reexport].............................
[package cl+ssl]..................................
[package dexador.encoding]........................
[package dexador.connection-cache]................
[package dexador.decoding-stream].................
[package dexador.keep-alive-stream]...............
[package dexador.error]...........................
[package dexador.util]............................
[package dexador.backend.usocket].................
[package dexador].................................
[package weblocks/dependencies]...................
[package weblocks/app-dependencies]...............
[package lack.response]...........................
[package lack.middleware.session.store]...........
[package lack.middleware.session.store.memory]....
[package lack.middleware.session.state]...........
[package lack.middleware.session.state.cookie]....
[package lack.middleware.session].................
[package weblocks/js/jquery]......................
[package weblocks/page]...........................
[package weblocks/session-lock]...................
[package weblocks/widgets/mop]....................
[package weblocks/widgets/dom]....................
[package cl-strings]..............................
[package weblocks/widgets/base]...................
[package weblocks/utils/warn].....................
[package trivial-backtrace].......................
[package weblocks/error-handler]..................
[package weblocks/utils/timing]...................
[package com.metabang.trivial-timeout]............
[package weblocks/widgets/render-methods].........
[package weblocks/widgets/root]...................
[package dissect].................................
[package log4cl-json.appender]....................
[package log4cl-json.core]........................
[package log4cl-json].............................
[package weblocks/request-handler]................
[package lack.component]..........................
[package lack.builder]............................
[package lack]....................................
[package lack.middleware.backtrace]...............
[package clack.util]..............................
[package clack.handler]...........................
[package clack]...................................
[package clack]...................................
[package weblocks/session-reset]..................
[package weblocks/debug]..........................
[package weblocks/server].........................
[package weblocks/widgets/string-widget]..........
[package weblocks/default-init]...................
[package weblocks/widgets/funcall-widget].........
[package weblocks/utils/clos].
(:WEBLOCKS)
CL-USER> (ql:where-is-system :WEBLOCKS)
#P"/private/tmp/check-weblocks-install/quicklisp/dists/ultralisp/software/40ants-weblocks-20181206174624/"
CL-USER> (ql:where-is-system :LACK-MIDDLEWARE-SESSION)
#P"/private/tmp/check-weblocks-install/quicklisp/dists/ultralisp/software/fukamachi-lack-20181109101309/"
```

Looks like the issue in your configuration. We need more info to investigate this problem.

**Comment by vindarel** [comment]
It doesn't work for me too. Before I told you on gitter, my build on CI was passing but the app didn't start, with this error. I retried the CI build and now [the build itself](https://gitlab.com/vindarel/cl-torrents-web/-/jobs/131709768) fails with this error.

> [package weblocks/server]Unhandled SB-KERNEL:SIMPLE-PACKAGE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                      {10005C05B3}>:
  The name "LACK-MIDDLEWARE-SESSION" does not designate any package.

**Comment by svetlyak40wt** [comment]
@vindarel you better to checkout a particular tag instead of doing:
```
cd ~/quicklisp/local-projects/ && git clone https://github.com/40ants/weblocks
Cloning into 'weblocks'...
```

because from the log of the build is impossible to understand which commit was checked out and lead to the error.

I always tag my libraries with tags like `v0.23.1` and write a ChangeLog, so it will be possible to pin concrete version in your app and upgrade intentionally, not occasionally :)

**Comment by svetlyak40wt** [comment]
Now I'll try to make a build of `cl-torrents-web` using a separate Dockerfile. Wait for pull request! :)

**Comment by svetlyak40wt** [comment]
Interesting. I tried to make a dockerfile based on `daewok/lisp-devel`, but `make install` does not work as expected, because this base image does some magic and unix user manipulation in it's `assets/entrypoint.sh` file: https://github.com/daewok/lisp-devel-docker/blob/af171fad47d1ff3a7aeff252b370b2a8063d4125/assets/entrypoint.sh

So, the simplest way, to recreate the same process on my machine is:

1. To create in the repository a file `simulate-ci.sh` with the following content:

```
#!/bin/bash

apt-get update -qy
apt-get install -y git-core
make install
make build
```

2. Run it inside the docker, using `daewok/lisp-devel` image:

```
docker run --rm -v `pwd`:/app -ti daewok/lisp-devel bash -c 'cd /app && ./simulate-ci.sh'
```

This way I was able to reproduce the problem:

```
[package weblocks/server]Unhandled SB-KERNEL:SIMPLE-PACKAGE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                      {10005C05B3}>:
  The name "LACK-MIDDLEWARE-SESSION" does not designate any package.

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10005C05B3}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<SB-KERNEL:SIMPLE-PACKAGE-ERROR "The name ~S does not designate any package." {1003C9C603}> #<unused argument> :QUIT T)
```

**Comment by svetlyak40wt** [comment]
I found the cause of the problem. It is a package and system's naming!

System `lack-middleware-session` defines a `lack.middleware.session` package, not `lack-middleware-session`. But because `weblocks` uses `package inferred system`, I've added this import, to fix @vindarel problem: https://github.com/40ants/weblocks/commit/e8b15e15c4f7b86157c35807bf56308b849dc510#diff-3d0afa1ed8c60bf294d17b2dc9a93462R9

But this import is wrong.
